source/_UIAHandler.py

```diff
@@ -538,6 +538,24 @@ def _isUIAWindowHelper(self,hwnd):
 		# allow the appModule for the window to also choose if this window is bad
 		if appModule and appModule.isBadUIAWindow(hwnd):
 			return False
+		if windowClass == "NetUIHWND":
+			# This class is used for various controls in MS Office.
+			# In versions of Office older than 2016 using IAccessible avoids some issues.
+			# Problems include lack of focus reporting (#4207), strange reporting of context menu items )#9252)
+			# and not being able to report ribbon sections when they starts with an edit  field (#7067)
+			# The last problem also exiists for Office 2016, however using IAccessible for collabsed ribbons
+			# causes NVDA not to report focus changes.
+			# FIX ME!
+			# Quick testing shows that prorer focus event are emitted yet they are ignored by NVDA.
+			if(
+				appModule.productName.startswith("Microsoft Office")

```

Comment:
why isn't this code in the appmodule as per `appModule.isBadUIAWindow` above? 

Also note, earlier lines indicate that appModule may be `None`

Reply:
Because all Office applications share the common product name but does not share single executable. Doing this at the appModule level would require separate module for at least:
- Word
- Excel
- Outlook
- Powerpoint
- Visio
- Project
- Access


and there is no  "guarantee that all Office apps are covered whereas using product name is safe.